### ISSUE TITLE
Multiple IService implementations causing 'System.DivideByZeroException' in ServiceRemotingMessageDispatcher

### ISSUE THREAD

**Issue by mbirkes** [microsoft/service-fabric-services-and-actors-dotnet#234]
When two or more IService implementations share the same interface name from different namespaces, and have at least one Operation with the same name, a System.DivideByZeroException occurs during the listener startup for any StatelessService that implements both IService interfaces.  

I originally posted this here, but realize it is actually a services SDK issue: https://github.com/Azure/service-fabric-issues/issues/1610

## Expected Behavior
Allow a single stateless service instance to implement multiple IService implementations. 

## Current Behavior
ServiceRemotingMessageDispatcher initializer throws exception: System.DivideByZeroException: 'Attempted to divide by zero.'  See stack trace in following section. 

## Steps to Reproduce
1. Define two IService interfaces with the same name in different namespaces
2. At least one method in each Interface should have the same operation Name (parameters can be different)
3. Create a StatelessService implementing the two interfaces
4. Setup a FabricTransportServiceRemotingListener for the service. 
5. During FabricTransportServiceRemotingListener constructor, the Exception will be thrown. 

The following sample code can reproduce the behavior. Removing/Renaming one of the 'OperationA' methods will allow the service to start successfully. 

```csharp
namespace MyServiceA
{
    public interface IMyInterface : IService
    {
        Task<string> OperationA(string request);
        Task OperationC(int request);
    }
}

namespace MyServiceB
{
    public interface IMyInterface : IService
    {
        Task<int> OperationA(int request);
        Task OperationD(int request);
    }
}

namespace MyService
{
   
    internal sealed class MyService : StatelessService, MyServiceA.IMyInterface, MyServiceB.IMyInterface
    {
        public MyService(StatelessServiceContext context) : base(context)
        {
        }

        protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceListeners()
        {
            return new[]
            {
                new ServiceInstanceListener(
                    context => new FabricTransportServiceRemotingListener(context, this), "MyServiceListener"),
            };
        }

        Task<string> MyServiceA.IMyInterface.OperationA(string request)
        {
            return Task.FromResult("test");
        }

        Task MyServiceA.IMyInterface.OperationC(int request)
        {
            return Task.CompletedTask;
        }

        Task<int> MyServiceB.IMyInterface.OperationA(int request)
        {
            return Task.FromResult(1);
        }

        Task MyServiceB.IMyInterface.OperationD(int request)
        {
            return Task.CompletedTask;
        }

    }
}
```

Stack trace:
`   at Microsoft.ServiceFabric.Services.Remoting.Diagnostic.PerformanceCounterInstanceNameBuilder.ComputeTruncatedMethodName(MethodNameBuilder methodNameBuilder)
   at Microsoft.ServiceFabric.Services.Remoting.Diagnostic.PerformanceCounterInstanceNameBuilder.ComputeMethodNameInCounterInstanceName(IEnumerable'1 methodNameBuilders, Dictionary'2 methodOverloads)
   at Microsoft.ServiceFabric.Services.Remoting.Diagnostic.PerformanceCounterInstanceNameBuilder.GetMethodCounterInstanceNames(IEnumerable'1 actorMethodInfo)
   at Microsoft.ServiceFabric.Services.Remoting.Diagnostic.ServicePerformanceCounterProvider.InitializeServiceMethodInfo()
   at Microsoft.ServiceFabric.Services.Remoting.Diagnostic.ServicePerformanceCounterProvider..ctor(Guid partitionId, Int64 replicaOrInstanceId, List'1 interfaceDescriptions, Boolean generateCounterforServiceCategory)
   at Microsoft.ServiceFabric.Services.Remoting.V2.Runtime.ServiceRemotingMessageDispatcher.Initialize(ServiceContext serviceContext, Object serviceImplementation, IEnumerable'1 remotedInterfaces, Boolean nonServiceInterface, IServiceRemotingMessageBodyFactory serviceRemotingMessageBodyFactory)
   at Microsoft.ServiceFabric.Services.Remoting.V2.Runtime.ServiceRemotingMessageDispatcher..ctor(ServiceContext serviceContext, IService serviceImplementation, IServiceRemotingMessageBodyFactory serviceRemotingMessageBodyFactory)
   at Microsoft.ServiceFabric.Services.Remoting.V2.FabricTransport.Runtime.FabricTransportServiceRemotingListener..ctor(ServiceContext serviceContext, IService serviceImplementation, FabricTransportRemotingListenerSettings remotingListenerSettings, IServiceRemotingMessageSerializationProvider serializationProvider)
   at MyService.MyService.<CreateServiceInstanceListeners>b__1_0(StatelessServiceContext context) in C:\Users\Mason.Birkes\source\repos\NewListenerTest\MyService\MyService.cs:line 53
   at Microsoft.ServiceFabric.Services.Runtime.StatelessServiceInstanceAdapter.<OpenCommunicationListenersAsync>d__20.MoveNext()`

## Context (Environment)
Need to support multiple interface implementation to the same service instance. This could include multiple listeners on the same service instance.  Currently some interfaces in different namespaces share Interface and Operation names and cannot currently be implemented on the same service with the current SDK. 

#### Service Fabric Runtime and SDK Version :
Service Fabric 7.0.457.9590
SDK: 4.0.457.9590
Microsoft.ServiceFabric.Services.Remoting : 4.0.457

#### Operating System :
Windows 10 1909 18363.476

#### Cluster Size :
One Node - Development 

## Possible Workaround
Create separate service instances for IServices with duplicated Operation names.

## Screenshot 
Sample image capturing the error
![Annotation 2019-11-29 180228](https://user-images.githubusercontent.com/272836/69961316-3c438600-14d1-11ea-8bed-9ec814e7c573.png)

**Comment by mbirkes** [comment]
The error occurs on line 254 of `PerformanceCounterInstanceNameBuilder.cs.` 

The `typeFull` variable is using the type Name instead of the FullName. Line 224:
`var typeFull = methodNameBuilder.MethodInfo.DeclaringType.Name;`

Changing the above line to the following line fixes the issues and the service can be hosted and interrogated by the client:
`var typeFull = methodNameBuilder.MethodInfo.DeclaringType.FullName;`

The error occurs because in `ComputeMethodNameInCounterInstanceName` the `methodOverloads` > 1 for the operation allowing flow to the `ComputeTruncatedMethodName`.  After applying the above fix, the namespace for the Facet is used and can successfully be injected into ` currentMethodBuilderInfo.Names[(int)MethodNameFormat.Truncated]` at line 211. 
See screenshot. 
![Annotation 2019-12-02 074548](https://user-images.githubusercontent.com/272836/69964487-25ecf880-14d8-11ea-8608-fad70366eaba.png)

**Comment by amanbha** [comment]
PR is merged, fix will be available with 4.2 SDK version (released with SF 7.2 runtime version)
